Lib/asyncio/futures.py

```diff
@@ -124,6 +126,20 @@ def get_loop(self):
             raise RuntimeError("Future object is not initialized.")
         return loop
 
+    def _make_cancelled_error(self):
+        """
+        Create the CancelledError to raise if the Future is cancelled.
+
+        This should only be called once when handling a cancellation since
+        it erases self._cancelled_exc.
+        """
+        exc = exceptions.CancelledError(
+            '' if self._cancel_message is None else self._cancel_message)
```

Comment:
Since we're back to having a constructor function for the `CancelledError` and in this case it's not easy to extract it, I'd suggest reverting to the old `if..ellse: exc = exceptions.CancelledError()` code.

Reply:
Thanks so much, @1st1. And on this point, I'm happy to make this change later tonight. I agree.